net/network.go

```diff
@@ -49,3 +56,8 @@ func (network *Network) GetPeerID() peer.ID {
 func (network *Network) GetBandwidthStats() metrics.Stats {
 	return network.Reporter.GetBandwidthTotals()
 }
+
+// Ping pings a peer and returns a channel with the amount of time for a response
+func (network *Network) Ping(ctx context.Context, pid peer.ID) (<-chan time.Duration, error) {
```

Comment:
you could avoid having this method if you wanted by directly embedding the ping service eg
```
type Network struct {
  ...
  *ping.PingService
}
```